services-core/src/solution_submission/retry.rs

```diff
@@ -138,13 +145,21 @@ impl<'a> RetryWithGasPriceIncrease<'a> {
         // recent one.
         // That is why we keep track of all past transaction futures in this variable.
         let mut futures = FuturesUnordered::new();
-        let mut highest_used_gas_price = 0.0;
+        let mut last_used_gas_price = 0.0;
         loop {
-            if let Some(gas_price) = self.new_gas_price_for_target_time(&args).await {
-                if gas_price >= highest_used_gas_price * MIN_GAS_PRICE_INCREASE_FACTOR {
-                    highest_used_gas_price = gas_price;
-                    futures.push(self.submit_solution(&args, gas_price));
+            match self.gas_price(&args).await {
+                Ok(gas_price) => {
+                    log::debug!("estimated gas price {}", gas_price);
+                    if let Some(mut gas_price) =
+                        calculate_gas_price(last_used_gas_price, gas_price, args.gas_price_cap)
+                    {
+                        gas_price = gas_price.ceil();
+                        last_used_gas_price = gas_price;
+                        log::info!("submitting solution transaction at gas price {}", gas_price);
```

Comment:
ubernit: I kind of liked that the old log message also gave us a hint if the current cap was hit (since this cannot be easily found from other informations in the logs). Maybe we can just add the cap to the log message